async-profiler: new port

oytech opened the pull request:
#### Description

new port for - https://github.com/async-profiler/async-profiler

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 14.7.5 23H527 arm64
Command Line Tools 16.2.0.0.1.1733547573

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] ~~referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message?~~ <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] ~~tried existing tests with `sudo port test`?~~
- [x] tried a full install with `sudo port -vs install`?
- [x] tested basic functionality of all binary files?
- [ ] ~~checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?~~

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten reviewed (changes requested):
MacPorts is intended to build packages from source, not to install pre-compiled binaries. Can this be changed so that it can be build instead of just downloading a binary from upstream its website, which an end-user can do easily themselves?

oytech commented:
> MacPorts is intended to build packages from source, not to install pre-compiled binaries. Can this be changed so that it can be build instead of just downloading a binary from upstream its website, which an end-user can do easily themselves?

I was not aware of that. Other port of java profiler VisualVM is installed from binary (does not work though). Sure I can look into that.

oytech commented:
@reneeotten I updated Portfile to build from source. It builds for me locally (`sudo port -vs install`). Can you approve build workflow?

reneeotten commented:
Since this port is not using a "configure" phase, you need to make sure it will respect the compiler selection/setting that MacPorts picks (see: [here](https://trac.macports.org/wiki/UsingTheRightCompiler). One convenient way to do that is to make use the `makefile` PortGroup. I have added that to the Portfile and pushed to your branch.

oytech commented:
> Since this port is not using a "configure" phase, you need to make sure it will respect the compiler selection/setting that MacPorts picks (see: [here](https://trac.macports.org/wiki/UsingTheRightCompiler). One convenient way to do that is to make use the `makefile` PortGroup. I have added that to the Portfile and pushed to your branch.

@reneeotten thanks for feedback. It makes sense. I updated the Portfile. Makefile of async-profiler uses following vars if set:
- `CC`, `CXX` (set by makefile.override)
- `JAVA_HOME` (set by PortGroup java)
- `CFLAGS_EXTRA`, `CXXFLAGS_EXTRA` (set manually in Portfile)

Is there anything else that should be fixed?

reneeotten commented:
I already fixed it by adding the "makefile" PortGroup so none of the changes you made are needed. I just messaged you as a heads up and to look over it and build it locally to make sure everything works.

oytech commented:
> I already fixed it by adding the "makefile" PortGroup so none of the changes you made are needed. I just messaged you as a heads up and to look over it and build it locally to make sure everything works.

I understand that `CC`, `CXX` are set by makefile PortGroup which I checked that upstream Makefile respects, but it does not respect `CFLAGS`, `CXXFLAGS` and uses alternative ones `CFLAGS_EXTRA`, `CXXFLAGS_EXTRA` which I added in the last changes. Am I missing something?

Yeah it builds and binary works.

oytech commented:
Also I added `xinstall` for other binary `jfrconv` which I missed before.

ryandesign commented:
This doesn't build on any system.

```
clang: error: no such file or directory: '=-fno-exceptions'; did you mean '-fno-exceptions'?
```

I don't see any CI results for this PR.

reneeotten commented:
> This doesn't build on any system.
> 
> ```
> clang: error: no such file or directory: '=-fno-exceptions'; did you mean '-fno-exceptions'?
> ```
> 
> I don't see any CI results for this PR.

my bad, fixed in bcd1c398cbb00d588a9eaba61871c9426c8708c7
